mlflow/utils/search_utils.py

```diff
@@ -2011,6 +2011,141 @@ def _get_comparison(cls, comparison):
         return comp
 
 
+@dataclass
+class TraceMetricsFilter:
+    view_type: str
+    entity: str
+    key: str | None
+    comparator: str
+    value: Any
+
+
+class SearchTraceMetricsUtils(SearchTraceUtils):
+    # Entities that are supported for each view type
+    # Mapping of entity to a boolean indicating if it requires a key
+    # For example, "tag" requires a key: "trace.tag.<key> = <value>"
+    # "status" does not require a key: "trace.status = <value>"
+    _VALID_TRACE_ENTITIES = {
+        "status": False,
+        "tag": True,
+        "metadata": True,
+    }
+    _VALID_SPAN_ENTITIES = {
+        "name": False,
+        "status": False,
+        "type": False,
+    }
+    _VALID_ASSESSMENT_ENTITIES = {
+        "name": False,
+        "type": False,
+    }
+    _VALID_VIEW_TYPES_TO_ENTITIES = {
+        "trace": _VALID_TRACE_ENTITIES,
+        "span": _VALID_SPAN_ENTITIES,
+        "assessment": _VALID_ASSESSMENT_ENTITIES,
+    }
+
+    @classmethod
+    def parse_search_filter_for_trace_metrics(cls, filter_string: str) -> TraceMetricsFilter:
+        parsed = cls.parse_search_filter(filter_string)
+        if len(parsed) != 1:
+            raise MlflowException.invalid_parameter_value(
+                f"Invalid filter: '{filter_string}'. Expected one filter clause."
+            )
+        return parsed[0]
+
+    @classmethod
+    def _process_statement(cls, statement: Statement) -> list[TraceMetricsFilter]:
```

Comment:
It's used in https://github.com/mlflow/mlflow/blob/a51c9a68f3b62dab240f3a4a615d05eb85b9b58f/mlflow/utils/search_utils.py#L521-L541